ngraph_bridge/ngraph_executor.cc

```diff
@@ -240,190 +220,167 @@ Status NGraphExecutor::GetNgExecutable(
-                     << input_tensors_bytes_free / (1024 * 1024) << " MB"
-                     << " Output Tensors freed: "
-                     << output_tensors_bytes_free / (1024 * 1024) << " MB";
-    }  // cache eviction if cache size greater than cache depth
-
-    ngraph::Event event_compile("Compile nGraph", m_node_name, "");
-    BackendManager::LockBackend(m_op_backend_name);
-    try {
-      if (m_do_aot) {
-        auto itr = m_aot_execs.find(signature);
-        if (itr == m_aot_execs.end()) {
-          BackendManager::UnlockBackend(m_op_backend_name);
-          return errors::Internal(
-              "Requested AOT, but could not find string with the "
-              "signature: ",
-              signature);
-        }
-        stringstream serialized_exec_read;
-        serialized_exec_read << (itr->second);
-        ng_exec = op_backend->load(serialized_exec_read);
-      } else {
-        ng_exec = op_backend->compile(ng_function);
-      }
-    } catch (const std::exception& exp) {
-      BackendManager::UnlockBackend(m_op_backend_name);
-      Status st = StringToFile("tf_function_error_" + m_node_name + ".json",
-                               serialized_ng_func);
-      string status_string =
-          "Caught exception while compiling op_backend: " + string(exp.what()) +
-          (st.ok() ? "" : (" Also error in dumping serialized function: " +
-                           st.error_message()));
-      return errors::Internal(status_string);
-    } catch (...) {
-      BackendManager::UnlockBackend(m_op_backend_name);
-      Status st = StringToFile("tf_function_error_" + m_node_name + ".json",
-                               serialized_ng_func);
-      string status_string =
-          "Error in compiling op_backend." +
-          (st.ok() ? "" : (" Also error in dumping serialized function: " +
-                           st.error_message()));
-      return errors::Internal(status_string);
+    serialized_ng_func = itr->second;
+  }
+
+  // Serialize to nGraph if needed
+  if (std::getenv("NGRAPH_ENABLE_SERIALIZE") != nullptr) {
+    std::string file_name = "tf_function_" + m_node_name + ".json";
+    auto status_ser = StringToFile("tf_function_" + m_node_name + ".json",
+                                   serialized_ng_func);
+    if (status_ser != Status::OK()) {
+      return std::make_pair(status_ser,
+                            std::make_tuple(ng_exec, serialized_ng_func, pts));
     }
-    BackendManager::UnlockBackend(m_op_backend_name);
-    event_compile.Stop();
-    ngraph::Event::write_trace(event_compile);
-    m_ng_exec_map[signature] = ng_exec;
-
-    // caching ng_function to serialize to ngraph if needed
-    m_serialized_ng_function_map[ng_exec] = serialized_ng_func;
-
-    m_lru.push_front(signature);
-    // Memory after
-    MemoryProfile(vm, rss);
-    auto delta_vm_mem = vm - vm0;
-    auto delta_res_mem = rss - rss0;
-    NGRAPH_VLOG(1) << "NGRAPH_TF_CACHE_PROFILE: OP_ID: " << m_instance_id
```

Comment:
We decided to get rid of the memory profiling?

Reply:
We added this for the LRU cache. But now we have a fixed size cache.

I am not sure if this is still effective or correct. We can keep this around I guess.

Reply:
Yes are not doing profiling anymore.